eudemo/eudemo/maintenance/equatorial_port.py

```diff
@@ -0,0 +1,229 @@
+    """
+    Castellation Builder parameters
+    """
+
+    n_components: Parameter[int]
+    cst_r_corner: Parameter[float]
+
+
+class EquatorialPortDesigner(Designer):
+    """
+    Equatorial Port Designer
+    """
+
+    param_cls: Type[EquatorialPortDesignerParams] = EquatorialPortDesignerParams
+
+    def __init__(
+        self,
+        params: Union[Dict, ParameterFrame, EquatorialPortDesignerParams],
+        build_config: Union[Dict, None],
+        koz_z_offset: float,
+        x_ib: float,
+        x_ob: float,
+    ):
+        super().__init__(params, build_config)
+        self.koz_offset = koz_z_offset
+        self.x_ib = x_ib
+        self.x_ob = x_ob
+
+    def run(self) -> BluemiraWire:
+        """
+        Design the xz keep-out zone profile of the equatorial port
+        """
+        x_ib = self.x_ib
+        x_ob = self.x_ob
+        z_h = self.params.ep_height.value + (2 * self.koz_offset)
+
+        x = (x_ib, x_ob, x_ob, x_ib)
+        z = (-z_h / 2, -z_h / 2, z_h / 2, z_h / 2)
+
+        ep_boundary = BluemiraFace(
+            make_polygon({"x": x, "y": 0, "z": z}, label="koz", closed=True)
+        )
+        return ep_boundary
+
+
+class CastellationBuilder(Builder):
+    """
+    Castellation Builder
+    """
+
+    NAME = "Castellation"
+    param_cls: Type[CastellationBuilderParams] = CastellationBuilderParams
+
+    def __init__(
+        self,
+        params: Union[Dict, ParameterFrame, CastellationBuilderParams],
+        build_config: Union[Dict, None],
+        length: float,
+        start_profile: BluemiraFace,
+        extrude_vec: tuple,
+        offsets: list,
+        castellation_offsets: list,
+    ):
+        super().__init__(params, build_config)
+        self.length = length
+        self.face = start_profile
+        self.vec = np.array(extrude_vec)
+        self.off = offsets
+        self.cst = castellation_offsets
+
+    def build(self) -> Component:
+        """
+        Build the castellated component.
+        """
+        # TODO: Implement corner radii using PR #1992
+        self.r_rad = self.params.cst_r_corner.value
+        self.n = self.params.n_components.value
+
+        xyz_solid = self.build_castellations(
+            self.face, self.vec, self.length, self.off, self.cst
+        )
+        xz_plane = BluemiraPlane(axis=(0, 1, 0))
+        xy_plane = BluemiraPlane(axis=(0, 0, 1))
+        xz_slice = BluemiraFace(slice_shape(xyz_solid, xz_plane))
+        xy_slice = BluemiraFace(slice_shape(xyz_solid, xy_plane))
+
+        return self.component_tree(
+            xz=[self.build_xz(xz_slice)],
+            xy=self.build_xy(xy_slice),
+            xyz=self.build_xyz(xyz_solid),
+        )
+
+    def build_xz(self, xz: BluemiraFace) -> PhysicalComponent:
+        """
+        Build the xz representation of the castellated component
+        """
+        body = PhysicalComponent(self.NAME, xz)
+        body.plot_options.face_options["color"] = BLUE_PALETTE["VV"][0]
```

Comment:
please use:

```python
from bluemira.builders.tools import apply_component_display_options
```
for this, it sets the color correctly in the plotting and the CAD